src/entity/User.ts

```diff
@@ -247,101 +223,120 @@ export const updateUserEmail = async (
 const isInfoConfirmed = (user: AddUserData) =>
   !!(user.name && user.email && user.username);
 
+const handleInsertError = async (
+  error: Error,
+  req: FastifyRequest,
+  con: DataSource | EntityManager,
+  data: DeepPartial<User>,
+  maxIterations = 5,
+  iteration = 0,
+): Promise<AddNewUserResult> => {
+  req.log.error(
+    {
+      data,
+      userId: data.id,
+      error,
+      iteration,
+      maxIterations,
+    },
+    'failed to create user profile',
+  );
+  const shouldRetry = iteration < maxIterations;
+  if ('code' in error) {
+    // Unique
+    if (error.code === TypeOrmError.DUPLICATE_ENTRY) {
+      if (error.message.indexOf('users_username_unique') > -1) {
+        return {
+          status: 'failed',
+          reason: UserFailErrorKeys.UsernameEmailExists,
+        };
+      }
+
+      if (error.message.indexOf('PK_') > -1) {
+        if (shouldRetry) {
+          data.id = await generateTrackingId(req, 'user creation');
+          return safeInsertUser(req, con, data, maxIterations, iteration + 1);
+        }
+        return { status: 'failed', reason: UserFailErrorKeys.UserExists };
+      }
+
+      // If it's not username or primary key than it's twitter and github.
+      if (shouldRetry) {
+        if (error.message.indexOf('users_twitter_unique') > -1) {
+          data.twitter = null;
```

Comment:
Are we not trying to solve this now in the wrong way then?
It means the user is already existing basically right, so would it not be better to feedback to the FE this is the case?

Reply:
Ah I see we already had this 🤔

Reply:
Yap it was this way all along just refactored a bit. The main fix is the id clashes